src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -86,15 +88,18 @@ public FunctionDescriptor GetHttpFunctionOrNull(Uri uri)
         protected override void OnInitializeConfig(JobHostConfiguration config)
         {
             base.OnInitializeConfig(config);
-
+            _metricsLogger = new WebHostMetricsLogger();
             // Add our WebHost specific services
-            config.AddService<IMetricsLogger>(new WebHostMetricsLogger());
+            config.AddService<IMetricsLogger>(_metricsLogger);
         }
 
         protected override void OnHostStarted()
         {
             base.OnHostStarted();
-
+            if (_metricsLogger != null)
+            {
+                _metricsLogger.OnHostStartedEvent(Instance);
```

Comment:
I was under the assumption that OnHostStarted will be called when a processes is started or file content changes and we want to raise ETW event in both these scenarios. What else can cause OnHostStarted() to get fired?